.github/workflows/dart.yml

```
@@ -88,34 +88,73 @@
+      - name: Run CI tests with coverage
         run: |
           echo "::group::Running tests for ${{ matrix.test-path }}"
-          dart test ${{ matrix.test-path }}
+          dart pub global activate coverage
+          dart test --coverage=coverage ${{ matrix.test-path }}
+          dart pub global run coverage:format_coverage --lcov --in=coverage --out=coverage/lcov.info --package=. --report-on=lib
           echo "::endgroup::"
 
-  generate-docs:
+      - name: Set artifact name
+        run: echo "ARTIFACT_NAME=coverage-$(echo '${{ matrix.test-path }}' | tr '/' '-')" >> $GITHUB_ENV
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:1:81: Double quote to prevent globbing and word splitting

(shellcheck)